rpcs3/Emu/RSX/VK/VKHelpers.h

```diff
@@ -320,30 +326,199 @@ namespace vk
 		}
 	};
 
-	struct memory_block
+	// Memory Allocator - base class
+
+	typedef void *HANDLE_DEV_MEM;	// abstract handle to device memory
```

Comment:
make this a scoped type `memory_allocator_base::mem_handle_t`. Add a `using mem_handle_t = void*;` in the public declarations of mem_allocator_base. This would keep its naming scheme consistent with the other types, we dont actually have to call it HANDLE_DEV_MEM. If the implementation requires it, define it in the implementation file (.cpp)